src/commands/createFunction/durableSteps/sql/SqlDatabaseListStep.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { Database, Server, SqlManagementClient } from '@azure/arm-sql';
+import { getResourceGroupFromId, uiUtils } from '@microsoft/vscode-azext-azureutils';
+import { AzureWizardExecuteStep, AzureWizardPromptStep, IAzureQuickPickItem, IAzureQuickPickOptions, ISubscriptionContext, IWizardOptions, nonNullProp, nonNullValue } from '@microsoft/vscode-azext-utils';
+import { localize } from '../../../../localize';
+import { createSqlClient } from '../../../../utils/azureClients';
+import { ISqlDatabaseConnectionWizardContext } from '../../../appSettings/ISqlDatabaseConnectionWizardContext';
+import { SqlDatabaseCreateStep } from './SqlDatabaseCreateStep';
+import { SqlDatabaseNameStep } from './SqlDatabaseNameStep';
+
+export class SqlDatabaseListStep<T extends ISqlDatabaseConnectionWizardContext> extends AzureWizardPromptStep<T> {
+    public async prompt(context: T): Promise<void> {
+        const client: SqlManagementClient = await createSqlClient(<T & ISubscriptionContext>context);
+        const rgName: string = getResourceGroupFromId(nonNullValue(context.sqlServer?.id));
+        const serverName: string = nonNullValue(context.sqlServer?.name);
+
+        const quickPickOptions: IAzureQuickPickOptions = { placeHolder: localize('selectSqlDatabase', 'Select a SQL database.') };
+        const picksTask: Promise<IAzureQuickPickItem<Database | undefined>[]> = this.getQuickPicks(uiUtils.listAllIterator(client.databases.listByServer(rgName, serverName)));
+
+        const result: Database | undefined = (await context.ui.showQuickPick(picksTask, quickPickOptions)).data;
+        context.sqlDatabase = result;
+    }
+
+    public async getSubWizard(context: T): Promise<IWizardOptions<T> | undefined> {
+        const promptSteps: AzureWizardPromptStep<T & ISubscriptionContext>[] = [];
+        const executeSteps: AzureWizardExecuteStep<T & ISubscriptionContext>[] = [];
+
+        if (context.sqlDatabase) {
+            context.valuesToMask.push(nonNullProp(context.sqlDatabase, 'name'));
+        } else {
+            promptSteps.push(new SqlDatabaseNameStep());
+            executeSteps.push(new SqlDatabaseCreateStep());
+        }
+
+        return { promptSteps, executeSteps };
+    }
+
+    public shouldPrompt(context: T): boolean {
```

Comment:
It does not.  This was something peculiar that I had to debug for a while because I thought that this was the case as well, but the execute steps happen all at the end even though there are previous `subwizard` steps to execute.